pkg/cloud/vsphere/provisioner/govmomi/create.go

```diff
@@ -292,8 +309,13 @@ func (pv *Provisioner) cloneVirtualMachine(s *SessionContext, cluster *clusterv1
 	// all.
 	diskMap := func(diskSpecs []vsphereconfigv1.DiskSpec) map[string]int64 {
 		diskMap := make(map[string]int64)
-		for _, s := range diskSpecs {
-			diskMap[s.DiskLabel] = s.DiskSizeGB
+		for i, s := range diskSpecs {
+			diskLabel := s.DiskLabel
+			if len(diskLabel) == 0 {
+				klog.Warningf("Missing label for VMDK. Using \"Hard disk %d\"", (i + 1))
+				diskLabel = fmt.Sprintf("Hard disk %d", (i + 1))
```

Comment:
I don't think we should do this. The original intent is for the user to be sure of what they want. Thus we want them the be precise about which disk out of their template they want to resize. Having this logic may lead to unintended behavior that the user may or may not have anticipated. Since this piece touches the data I prefer that the user to be more explicit in their choice rather than us making some assumptions.

Reply:
Makes sense... how about if there is only a single VMDK, we attempt to use hard disk label. If the label does not match even using the default value, then the deployment is going to fail anyways.

Reply:
The point in the `disks` section of the spec is optional.. the user if they don't want to make any disk changes they can simply leave that section out and that would work just fine.

For the Network part, yes it will fail. But the proper way is we need to report it back on the machineStatus to let the user know what went wrong, or what was the missing info. We haven't implemented that yet, but it would be great if you can add that as the default mechanism of giving the user feedback on what went wrong!

Reply:
the providerSpec required and optional items are not been documented, it is in api code, but not obvious to users, we need to improve on this. (obvious not for this PR)

Reply:
yea @figo that's what I was kinda working around.

By default, when you generate the `./out` files, it creates the yaml for both the disks and network fields. I ended up getting stuck for a bit because this was not immediately obvious without a lot of trial and error.

This was an attempt to  use the default generated `./out` files and also give the user experience a "it just works" type feel. Either way, I can remove both code blocks. Just let me know what the consensus is.

Reply:
Sidharth has a good point.  When I saw this, I just thought of it as more validation logic.  We should remove this.

Reply:
no problem. will remove it.